--- a/javascript/net/grpc/web/generator/grpc_generator.cc
+++ b/javascript/net/grpc/web/generator/grpc_generator.cc
@@ -865,8 +865,8 @@ void PrintProtoDtsMessage(Printer* printer, const Descriptor* desc,
     printer->Print("\n");
   }
 
-  for (int i = 0; i < desc->real_oneof_decl_count(); i++) {
-    const OneofDescriptor *oneof = desc->real_oneof_decl(i);
+  for (int i = 0; i < desc->oneof_decl_count(); i++) {
+    const OneofDescriptor *oneof = desc->oneof_decl(i);
     vars["js_oneof_name"] = ToUpperCamel(ParseLowerUnderscore(oneof->name()));
     printer->Print(
         vars, "get$js_oneof_name$Case(): $class_name$.$js_oneof_name$Case;\n");